src/main.rs

```diff
@@ -9,12 +9,51 @@
 
 #[macro_use] extern crate log;
 #[macro_use] extern crate clap;
+#[macro_use] extern crate error_chain;
 extern crate git2;
+extern crate libgitdit;
+
+mod error;
 
 use clap::App;
+use git2::Repository;
+use libgitdit::repository::RepositoryExt;
+
+use error::ErrorKind as EK;
+use error::*;
+
+
+/// Open the DIT repo
+///
+/// Opens the DIT repo corresponding to the current one honouring the user
+/// configuration.
+///
+fn open_dit_repo() -> Result<Repository> {
+    // TODO: access the config and maybe return another repo instead
+    Repository::open_from_env().chain_err(|| EK::WrappedGitError)
+}
+
+
+/// find-tree-init-hash subcommand implementation
+///
+fn find_tree_init_hash(repo: &Repository, matches: &clap::ArgMatches) -> Result<()> {
+    repo.revparse_single(matches.value_of("commit").unwrap())
+        .and_then(|obj| repo.find_commit(obj.id()))
+        .chain_err(|| EK::WrappedGitError)
+        .and_then(|commit| repo.find_tree_init(commit).chain_err(|| EK::WrappedGitDitError))
+        .map(|commit| println!("{}", commit.id()))
+}
+
 
 fn main() {
     let yaml    = load_yaml!("cli.yaml");
     let matches = App::from_yaml(yaml).get_matches();
-    println!("Hello, world!");
+
+    let repo = open_dit_repo().unwrap();
+
+    match matches.subcommand() {
+        ("find-tree-init-hash", Some(sub_matches))  => find_tree_init_hash(&repo, sub_matches),
+        (name, _) => Err(Error::from_kind(EK::SubCommandNotFound(String::from(name)))),
+        // TODO: find and exec 3rd party executable
+    }.unwrap()
```

Comment:
Yep. Originally, I planned on implementing proper error handling later on. But I'll implement the changes in this PR. This will also simplify implementing #28.